code/datums/components/surgery_initiator.dm

```diff
@@ -0,0 +1,278 @@
+/// Allows an item to be used to initiate surgeries.
+/datum/component/surgery_initiator
+	/// If present, this surgery TYPE will be attempted when the item is used.
+	/// Useful for things like limb reattachments that don't need a scalpel.
+	var/datum/surgery/forced_surgery
+
+	/// If true, the initial step will be cancellable by just using the tool again. Should be FALSE for any tool that actually has a first surgery step.
+	var/can_cancel_before_first = FALSE
+
+	/// If true, can be used with a cautery in the off-hand to cancel a surgery.
+	var/can_cancel = TRUE
+
+/datum/component/surgery_initiator/Initialize(datum/surgery/forced_surgery = null)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.forced_surgery = forced_surgery
+
+/datum/component/surgery_initiator/Destroy(force, silent)
+	return ..()
+
+/datum/component/surgery_initiator/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK, .proc/initiate_surgery_moment)
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_SHARPNESS, .proc/on_parent_sharpness_change)
+
+/datum/component/surgery_initiator/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_ATTACK)
+	UnregisterSignal(parent, COMSIG_ATOM_UPDATE_SHARPNESS)
+
+/// Keep tabs on the attached item's sharpness. Any sharp item can start a surgery.
+/datum/component/surgery_initiator/proc/on_parent_sharpness_change()
+	SIGNAL_HANDLER  // COMSIG_ATOM_UPDATE_SHARPNESS
+	var/obj/item/P = parent
+	if(!P.sharp)
+		RemoveComponent()
+
+/// Does the surgery initiation.
+/datum/component/surgery_initiator/proc/initiate_surgery_moment(datum/source, atom/target, mob/user)
+	SIGNAL_HANDLER	// COMSIG_ITEM_ATTACK
+	if(!isliving(user))
+		return
+	if(!user.Adjacent(target))
+		return
+	if(user.a_intent != INTENT_HELP)
+		return
+	var/mob/living/L = user
+	if(istype(L) && L.has_status_effect(STATUS_EFFECT_SUMMONEDGHOST))
+		to_chat(user, "<span class='notice'>You realise that a ghost probably doesn't have any useful organs.</span>")
+		return //no cult ghost surgery please
+	INVOKE_ASYNC(src, .proc/do_initiate_surgery_moment, target, user)
+	return TRUE
+
+/// Meat and potatoes of starting surgery.
+/datum/component/surgery_initiator/proc/do_initiate_surgery_moment(mob/living/target, mob/user)
+	var/datum/surgery/current_surgery
+
+	// Check if we've already got a surgery on our target zone
+	for(var/i_one in target.surgeries)
+		var/datum/surgery/surgeryloop = i_one
+		if(surgeryloop.location == user.zone_selected)
+			current_surgery = surgeryloop
+			break
+
+	if(!isnull(current_surgery) && !current_surgery.step_in_progress)
+		var/datum/surgery_step/current_step = current_surgery.get_surgery_step()
+		if(current_step.try_op(user, target, user.zone_selected, parent, current_surgery))
+			return
+		if(istype(parent, /obj/item/scalpel/laser/manager/debug))
+			return
+		if((can_cancel_before_first && current_surgery.status == 1) || current_surgery.status > 1)
```

Comment:
is this 1 meant to be `TRUE` , `SURGERY_STEP_CONTINUE` or something else?

Reply:
Surgery status is just the current step number that it's on. I could probably change the var name to be a bit more accurate

Reply:
Addressed in https://github.com/ParadiseSS13/Paradise/pull/18325/commits/37743ca333fc38b330b297d4dee783bf3d6e8388. It's now `step_number`.